### ISSUE TITLE
Only lint `node` .js files? (and not browser .js files)

### ISSUE THREAD

**Issue by foxtree** [mysticatea/eslint-plugin-node#88]
Is it possible to configure ESLint, such that it will only check `eslint-plugin-node` rules when the `node` environment is enabled?  

For example, `eslint-plugin-node` rules would be applied if the following directive existed:
(for files that execute in a Node.js context)

```
/* eslint-env node */
```

**Comment by jordanbtucker** [comment]
I would like to add that the rules should only apply when `{env: {node: true}}`.

Use case:

I have a project with backend and frontend code with this folder structure:

```
app
|- .eslintrc.json
|- server.js
|- /public
  |- .eslintrc.json
  |- client.js
  |- index.html
```

`/.eslintrc.json`:

```json
{
  "env": {
    "node": true
  },
  "plugins": {
    "node"
  },
  "extends": [
    "plugin:node/recommended"
  ]
}
```

`/frontend/.eslintrc.json`:

```json
{
  "env": {
    "browser": true,
    "node": false
  }
}
```

`/frontend/client.js`:

```js
import _ from 'lodash' // <- node/no-unsupported-features complains here
```

`eslint-plugin-node` shouldn't care about unsupported features (or any other rules) when `node: false`.

**Comment by mysticatea** [comment]
Thank you for this proposal.

However, I don't think it's feasible.
Plugin rules cannot get the value of `env` which is defined by config files. 
Also, there are people which don't use `env` to avoid defining unnecessary globals.

I'm closing as won't fix.

**Comment by nfriedly** [comment]
Hey, I know this is a really old issue, but I ran into the same problem today and I was wondering if anything has changed in the last few years?
